crates/provider/src/utils.rs

```diff
@@ -48,6 +53,63 @@ pub fn eip1559_default_estimator(base_fee_per_gas: U256, rewards: &[Vec<U256>])
     (potential_max_fee, max_priority_fee_per_gas)
 }
 
+/// Identifies the intended transport type from the given string.
+pub fn parse_str_to_tranport_type(
+    s: &str,
+) -> Result<BuiltInTransportType, RpcError<TransportErrorKind>> {
+    // Parse if string is a socket address e.g. "127.0.0.1:8545"
+    let s = s.parse::<SocketAddr>().ok().map_or_else(|| s.to_string(), |s| format!("http://{}", s));
+
+    // Check if s is a path and it exists
+    let path = Path::new(&s);
+    if path.exists() && path.is_file() {
+        // IPC if it exists
+        return Ok(BuiltInTransportType::Ipc);
+    }
+
+    // Parse the URL or return an error
+    Url::parse(&s).map_err(|e| TransportErrorKind::custom_str(e.to_string().as_str()))?;
+
+    // Return the transport type or error
+    BuiltInTransportType::from_str(&s)
+        .map_err(|_| TransportErrorKind::custom_str("Invalid transport type"))
+}
+
+/// The built-in transport types.
+#[derive(Debug, PartialEq, Eq)]
+pub enum BuiltInTransportType {
+    /// HTTP transport.
+    Http(String),
```

Comment:
ideally, each variant should hold the authenticated connection information. We may also want to implement a `fn connect(self) -> BoxTransport` for this. The ideal experience here would be

```
let transport: BoxTransport = my_str.parse::<BuiltinTransportDetails>()?.connect();